src/pages/UserSettingsPage/UserSettingsPage.tsx

```diff
@@ -0,0 +1,283 @@
+              formState,
+            }) => (
+              <TextField
+                label="User Picture"
+                variant="outlined"
+                sx={{ mb: 2, width: '100%' }}
+                inputRef={ref}
+                helperText={
+                  !!error && isTouched && <>URL to picture. May be blank.</>
+                }
+                error={!!error && isTouched}
+                onChange={onChange}
+                onBlur={onBlur}
+                placeholder="https://example.com/picture.jpg"
+                value={value}
+                name={name}
+              />
+            )}
+            rules={{
+              required: false,
+              pattern: /^https?:\/\/.*\.(?:png|jpg|jpeg|gif|svg|webp)$/
+            }}
+          />
+          <Controller
+            control={control}
+            name="user_bio"
+            render={({
+              field: { ref, onChange, onBlur, value, name },
+              fieldState: { isTouched, isDirty, error },
+              formState,
+            }) => (
+              <TextField
+                label="User Bio"
+                variant="outlined"
+                multiline
+                minRows={2}
+                sx={{ mb: 2, width: '100%' }}
+                inputRef={ref}
+                helperText={
+                  !!error && isTouched && <>Please say something about yourself.</>
+                }
+                error={!!error && isTouched}
+                onChange={onChange}
+                onBlur={onBlur}
+                value={value}
+                name={name}
+              />
+            )}
+            rules={{ required: true }}
+          />
+          <Controller
+            control={control}
+            name="zipcode"
+            render={({
+              field: { ref, onChange, onBlur, value, name },
+              fieldState: { isTouched, isDirty, error },
+              formState,
+            }) => (
+              <TextField
+                label="Zip Code"
+                variant="outlined"
+                sx={{ mb: 2, width: '100%' }}
+                inputRef={ref}
+                placeholder={"20036"}
+                helperText={
+                  !!error && isTouched && <>Please enter a valid zip code.</>
+                }
+                error={!!error && isTouched}
+                onChange={onChange}
+                onBlur={onBlur}
+                value={value}
+                name={name}
+              />
+            )}
+            rules={{
+              required: true, 
+              pattern: /^\d{5}(?:[-\s]\d{4})?$/
+            }}
+          />
+          <FormControl fullWidth sx={{ mb: 2 }}>
+            <InputLabel id="max_distance">Max Distance (miles)</InputLabel>
+            <Select labelId="max_distance"
+              id="max_distance"
+              label="Max Distance (miles)"
+            >
+              <MenuItem value={2}>2 miles</MenuItem>
```

Comment:
Would creating numbers of arrays, and map them to return JSX be cleaner/ more efficient in this case? 🤔  I just thought it might look cleaner that way. Could be in for the future update or something